### PR TITLE
GUI-2122 Bucket Upload Object

### PR DESCRIPTION
**Pull request by coyote240**
https://eucalyptus.atlassian.net/browse/GUI-2122

PR is getting pretty big, so let's checkpoint here.  Also contains elements of GUI-2121, GUI-2123 and GUI-2124.  tests/buckets.py script will run through scenarios leaving no lingering artifacts.

Contains some stubs for next iteration.

### CODE REVIEW

File: eucaconsole/templates/buckets/bucket_contents.pt [hunk 1]
```diff
@@ -112,10 +112,10 @@
                         <a i18n:translate="" ng-click="doPaste('${bucket_name}', item)">Paste folder</a>
                     </li>
                     <li ng-show="item.is_folder">
-                        <a i18n:translate="" ng-click="revealModal('delete-folder', item)">Delete</a>
```
Comment (coyote240) [inline eucaconsole/templates/buckets/bucket_contents.pt]: Beginning the theme of adding semantic CSS classes for DOM navigation.

File: eucaconsole/templates/dialogs/bucket_item_dialogs.pt [hunk 2]
```diff
@@ -23,7 +23,7 @@
             <strong>{{ obj_key }}</strong>
             <span i18n:translate="">public?</span>
         </p>
-        <a class="button expand" id="delete-object-btn" i18n:translate="" ng-click="makeObjectPublic()">
```
Comment (coyote240) [inline eucaconsole/templates/dialogs/bucket_item_dialogs.pt]: Oops, two of the same ID on one page!  Reliance on ids is dangerous, even for testing.  Will begin demonstrating more class-based selectors in the future.

**Comment by aliceh** [comment]
@coyote240 I could run the test on console built from packages! Thanks for the fix!

**Comment by aliceh** [comment]
Retest this, please

**Comment by aliceh** [comment]
Retest this, please.

**Comment by aliceh** [comment]
Build passed: http://jenkins.qa1.eucalyptus-systems.com/view/Eucaconsole%20CI/job/user-console-PR-monitor/1127/

**Comment by aliceh** [comment]
@coyote240 good to merge into master and maint branches!

File: eucaconsole/static/js/pages/bucket_contents.js [hunk 3]
```diff
@@ -33,7 +33,7 @@ angular.module('BucketContentsPage', ['LandingPage', 'EucaConsoleUtils'])
             $scope.makeObjectPublicUrl = options.make_object_public_url;
             // set upload button target based on media query
             if (Foundation.utils.is_medium_up()) {
-                $('#upload-file-btn').attr('target', '_blank');
+                $('#upload-file-btn').attr('target', 'upload-file');
```
Comment (dkavanagh) [inline eucaconsole/static/js/pages/bucket_contents.js]: The goal is to open a branch new page. Does this still do that?

**Pull request merged**